File: src/help_function/help_function.cc
#include "help_function.h"

int find_coprime(int a) {
    for (int i = 2; i < a; i++) {
        if (gcd(i, a) == 1) {
            return i;
        }
    }
    return -1; // возвращаем -1 если такое число не найдено
}

int find_inverse(int a, int m) {
    int x = 0, y = 0;
    int gcd = gcdExtended(a, m, &x, &y);
    if (gcd != 1) {
        std::cout << "Inverse doesn't exist" << std::endl;
        return -1;
    } else {
        return (x % m + m) % m;
    }
}

int pow_mod_10(int num, int pow) {
    num %= 10;
    int result = 1;
    for (int i = 0; i < pow; i++) {
        result *= num;
        result %= 10;
    }
}

File: src/Python/help_function.py
import math 
from Q4_gcdExtended import gcdExtended

def find_coprime(a):
    for i in range(2, a):
        if math.gcd(i, a) == 1:
            return i
    return -1

def find_inverse(a, m):
    x, y = 0, 0
    gcd, x, y = gcdExtended(a, m, x, y)
    if gcd != 1:
        print("Inverse doesn't exist")
        return -1
    else:
        return (x % m + m) % m

def pow_mod_10(num, pow):
    num %= 10
    result = 1
    for i in range(pow):
        result *= num
        result %= 10